Hah. Knew I forgot something (literally did not even try to test it)

I can on principle pull the I connect stuff out, but I think in that case
we'd need a blocking variant of connect() to make it useful again (since
there's stuff you don't want to do until you're connected )

On Thursday, February 5, 2015, Mike Ryan notifications@github.com wrote:

> I'm good with this in principle and I think it's the Right Thing. I ran
> into one issue trying to run it:
> 
> Traceback (most recent call last):
>   File "/usr/local/bin/gatt_client", line 9, in <module>
>     load_entry_point('PyBT==0.1', 'console_scripts', 'gatt_client')()
>   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 351, in load_entry_point
>     return get_distribution(dist).load_entry_point(group, name)
>   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2363, in load_entry_point
>     return ep.load()
>   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2088, in load
>     entry = **import**(self.module_name, globals(),globals(), ['**name**'])
>   File ".../PyBT/PyBT/gattclient.py", line 138, in <module>
>     'onconnect': CommandModule.onconnect,
> AttributeError: type object 'CommandModule' has no attribute 'onconnect'
> 
> Looks like you simply forgot to move onconnect into gatt_core. It seems
> like the onconnect stuff belongs in gattclient proper (since it relates
> mostly to an interactive client). There are some onconnect hooks in the
> guts of gatt_core, not sure what the right way to move these into
> gattclient is.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mikeryan/PyBT/pull/1#issuecomment-72998178.

--- a/PyBT/gatt_core.py
+++ b/PyBT/gatt_core.py
@@ -4,6 +4,10 @@ import select
 import functools
 import threading
 import gevent
+from gevent.select import select
+# this is hack because the above does not work
+from gevent import monkey
+monkey.patch_select()
 
 from PyBT.gap import GAP
 from PyBT.stack import BTEvent

--- a/PyBT/gatt_core.py
+++ b/PyBT/gatt_core.py
@@ -100,7 +100,7 @@ class Connection(object):
 
     def _dispatchSocketHandler(self):
         handler = SocketHandler(self)
-        gevent.spawn(handler.start)
+        gevent.spawn(handler.run)
 
     # Public command functions
 

--- a/PyBT/gattclient.py
+++ b/PyBT/gattclient.py
@@ -8,18 +8,6 @@ from binascii import unhexlify
 from PyBT.roles import LE_Central
 from PyBT.gatt_core import Connection, ConnectionError
 
-from gevent.select import select
-# this is hack because the above does not work
-from gevent import monkey
-monkey.patch_select()
-
-DISCONNECTED = 0
-CONNECTED = 1
-
-central = None
-state = DISCONNECTED
-onconnect = []
-
 def debug(msg):
     if os.getenv("DEBUG"):
         sys.stdout.write(msg)
@@ -135,7 +123,6 @@ COMMANDS = {
     'write-cmd': CommandModule.write_cmd,
     'read': CommandModule.read,
     'interval': CommandModule.interval,
-    'onconnect': CommandModule.onconnect,
 }
 
 

